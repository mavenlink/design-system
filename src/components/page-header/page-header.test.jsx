import React from 'react';
import { render, screen } from '@testing-library/react';
import PageHeader from './page-header.jsx';

describe('PageHeader', () => {
  const requiredProps = {
    title: 'Title',
  };

  it('has defaults', () => {
    render(<PageHeader {...requiredProps} />);
    expect(document.body).toMatchSnapshot();
  });

  it('has a title', () => {
    render(<PageHeader {...requiredProps} title="Another title" />);
    expect(screen.getByText('Another title').tagName).toEqual('H1');
  });

  it('has a description', () => {
    render(<PageHeader {...requiredProps} description="I am a description" />);
    expect(screen.getByText('I am a description').tagName).toEqual('P');
  });

  it('has a className API', () => {
    render(<PageHeader {...requiredProps} className="test-class" />);
    expect(screen.getByText('Title').parentElement).toHaveClass('test-class');
  });
});
