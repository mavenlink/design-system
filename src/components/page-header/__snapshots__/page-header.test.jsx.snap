// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader has defaults 1`] = `
<body>
  <div>
    <header
      class="container"
    >
      <h1
        class="heading"
      >
        Defaults
      </h1>
    </header>
  </div>
</body>
`;
